usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -539,6 +560,57 @@
         global_temp.ref_country_code AS rcc
             ON (rcc.country_code = UPPER(bs.sub_legal_entity_country_code)
                 AND bs.sub_legal_entity_country_code IS NOT NULL)
+    LEFT OUTER JOIN
+        global_temp.cd_state_grouped pop_cd_state_grouped ON (
+            pop_cd_state_grouped.state_abbreviation=UPPER(bs.sub_place_of_perform_state_code)
+            AND pop_cd_state_grouped.congressional_district_no <> '90'
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_state_grouped rl_cd_state_grouped ON (
+            rl_cd_state_grouped.state_abbreviation=UPPER(bs.sub_legal_entity_state_code)
+            AND rl_cd_state_grouped.congressional_district_no <> '90'
+        )
+    LEFT OUTER JOIN
+        raw.zips pop_zips ON (
+            pop_zips.zip5=LEFT(COALESCE(sub_place_of_performance_zip, ''), 5)
+            AND pop_zips.zip_last4=RIGHT(COALESCE(sub_place_of_performance_zip, ''), 4)
+        )
+    LEFT OUTER JOIN
+        raw.zips rl_zips ON (
+            rl_zips.zip5=LEFT(COALESCE(bs.sub_legal_entity_zip, ''), 5)
+            AND rl_zips.zip_last4=RIGHT(COALESCE(bs.sub_legal_entity_zip, ''), 4)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_zips_grouped pop_cd_zips_grouped ON (
+            pop_cd_zips_grouped.zip5=LEFT(COALESCE(sub_place_of_performance_zip, ''), 5)
+            AND pop_cd_zips_grouped.state_abbreviation=UPPER(bs.sub_place_of_perform_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_zips_grouped rl_cd_zips_grouped ON (
+            rl_cd_zips_grouped.zip5=LEFT(COALESCE(bs.sub_legal_entity_zip, ''), 5)
+            AND rl_cd_zips_grouped.state_abbreviation=UPPER(bs.sub_legal_entity_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_city_grouped pop_cd_city_grouped ON (
+            pop_cd_city_grouped.city_name=UPPER(bs.sub_place_of_perform_city_name)
+            AND pop_cd_city_grouped.state_abbreviation=UPPER(bs.sub_place_of_perform_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_city_grouped rl_cd_city_grouped ON (
+            rl_cd_city_grouped.city_name=UPPER(bs.sub_legal_entity_city_name)
+            AND rl_cd_city_grouped.state_abbreviation=UPPER(bs.sub_legal_entity_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_county_grouped pop_cd_county_grouped ON (
+            pop_cd_county_grouped.county_number=LPAD(CAST(CAST(REGEXP_EXTRACT(pop.county_numeric, '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 3, '0')
```

Comment:
This regex has appeared three times now. It may be a good idea to store in a variable so that it does not need to be duplicated.